curriculum/challenges/english/20-upcoming-python/learn-list-comprehension-by-building-a-case-converter-program/657efdcf7fe23b76c0cff9ec.md

```diff
@@ -0,0 +1,41 @@
+---
+id: 657efdcf7fe23b76c0cff9ec
+title: Step 7
+challengeType: 20
+dashedName: step-7
+---
+
+# --description--
+
+Add an `else` clause on the same level as the existing `if` statement, inside the `for` loop. Add characters that are already in lowercase to the list of converted characters inside the body of the `else` clause.
+
+# --hints--
+
+You should add an `else` clause inside the `for` loop. Don't forget the colon at the end.
+
+```js
+({
+    test: () => {
+        const transformedCode = e.code.original["main.py"].replace(/\r/g, "");
+        const convert_to_snake_case = __helpers.python.getDef("\n" + transformedCode, "convert_to_snake_case");
+        const { function_body } = convert_to_snake_case;
+
+        assert.match(function_body, / +else:/);
+    }
+})
+```
+
```

Comment:
Remember to add test for the `else` body.

Reply:
Just added the test.